### DISCUSSION TITLE
How do I define a macro at a custom theme level so it can be reused by multiple projects using this theme?

### DISCUSSION THREAD

**Discussion by ssbarnea** [fralau/mkdocs-macros-plugin#218] [Q&A]
We have something like 10 projects that all rely on a custom mkdocs theme and I want to add a macro that would expand in each of this projects but without having to duplicate its definition in each of them.

How can I achieve this?

PS. This might be related to https://github.com/squidfunk/mkdocs-material/discussions/6911

**Comment by fralau** [comment]
We need to make sure we are talking about the same thing? MkDocs-Macros is a **plugin**, which is distinct from a theme. However, it is possible that one would want to make a macro specifically for the Material theme.

In any case, it is understandable that you want to reuse the same macros across different projects. To avoid duplicating code, the solution is to make your module file into a [pluglet](https://mkdocs-macros-plugin.readthedocs.io/en/latest/pluglets/). A pluglet is basically a module installable through `pip install`. After that you declare it in the config file, and you are good to go.

**Comment by fralau** [comment]
@ssbarnea How did it go? Did you manage to achieve the result you want?
